histomicstk/segmentation/label/isbfcpp.h

```diff
@@ -0,0 +1,42 @@
+/*
+C++ version of ISBF for TraceBounds
+*/
+#ifndef isbfcpp_h
+#define isbfcpp_h
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+struct node
+{
+    int val_x;
+    int val_y;
+    node *next;
+};
+
+class isbfcpp
+{
+private:
+    node *head;
+    node *curr;
+    int count;
+    int nrows;
+    int ncols;
+    int** matrix00;
+    int** matrix90;
+    int** matrix180;
+    int** matrix270;
+
+public:
+    isbfcpp();
+    int length();
+    void nth_from_last(int n, int &x, int &y);
+    bool addList(int x, int y);
```

Comment:
Use std::vector for x and y coords of the boundary. 

Then you wont need the `addList` (use pushback function of std::vector) and `nth_from_last` (you can just index in the vector and get any element) functions.